Extract fakeHubspotApi

MattMSumner opened the pull request:
Why:

The tests are becoming noisey with all of the endpoint mocking.

This PR:

Extracts a single file to mock all of Hubspot's api and use that in our
tests.

SViccari commented on test/helpers/fake_hubspot_api.js:
```diff
@@ -0,0 +1,105 @@
+const nockHelper = require('./nock_helper')
+const applicationId = process.env.APPLICATION_ID || 12345
+const appId = process.env.APP_ID || 23456
+const userId = process.env.USER_ID || 23456
```
This is minor but I think it would be helpful to use a different value for the `appId` and the `userId` so there isn't a false sense of similarity between these two values.

SViccari commented on test/helpers/nock_helper.js:
```diff
@@ -1,11 +1,30 @@
+  statusCode = 200,
+}) => {
+  nock('http://api.hubapi.com', { encodedQueryParams: true })
+    // .log(console.log)
```
Can this commented out code be removed?

SViccari commented on test/campaigns.js:
```diff
@@ -37,21 +36,23 @@ describe('campaigns', function() {
+        response: { id: campaignId },
+      }
+      fakeHubspotApi.setupServer({
+        demo: true,
```
I'm uncertain why the `hapikey` needs to be set for this endpoint but not for the other endpoint, like `campaignsGetEndpoint`. Can you shed some light on the use of `demo: true`?

MattMSumner replied:
Sure. I couldn't work out how to get email campaigns on my testing server so I decided to leave this on the demo account.

SViccari reviewed (approved):
I like how this introduces the `fakeHubspotAPI` abstraction instead of using nock directly in each test 👍 Nice work, this LGTM 🐳

MattMSumner committed:
Extract fakeHubspotApi

Why:

The tests are becoming noisy with all of the endpoint mocking.

This PR:

Extracts a single file to mock all of Hubspot's api and use that in our
tests.
